### PR TITLE
[Python] Fix crash if no valid client define in typespec file

### PR DESCRIPTION
**Pull request by msyyc**
fix https://github.com/Azure/autorest.python/issues/2933

- Fix crash if no valid client define in typespec file
- Pad special property name in model to avoid conflict

### CODE REVIEW

File: packages/http-client-python/emitter/src/emitter.ts [hunk 1]
```diff
@@ -78,6 +78,12 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
   const root = path.join(dirname(fileURLToPath(import.meta.url)), "..", "..");
   const outputDir = context.emitterOutputDir;
   const yamlMap = emitCodeModel(sdkContext);
+  if (yamlMap.clients.length === 0) {
+    // eslint-disable-next-line no-console
+    console.log("Warning: Can't generate Python SDK since there is no valid client defined!!!");
+    return;
```
Comment (msyyc) [inline packages/http-client-python/emitter/src/emitter.ts:84]: Python emitter generate SDK started from client, if no clients define, we shall stop generation and log warning.

Comment (tadelesh) [reply packages/http-client-python/emitter/src/emitter.ts:84]: we do not use console to log error before. it is possible to generate model only packages. i suggest to just return here.

Comment (msyyc) [reply packages/http-client-python/emitter/src/emitter.ts:86]: Make sense.

Comment (msyyc) [reply packages/http-client-python/emitter/src/emitter.ts:86]: Updated to use reportDiagnotics to report warning.

File: packages/http-client-python/emitter/src/lib.ts [hunk 2]
```diff
@@ -49,7 +49,14 @@ const EmitterOptionsSchema: JSONSchemaType<PythonEmitterOptions> = {
 
 const libDef = {
   name: "@typespec/http-client-python",
-  diagnostics: {},
+  diagnostics: {
+    "no-valid-client": {
+      severity: "warning",
```
Comment (tadelesh) [inline packages/http-client-python/emitter/src/lib.ts:54]: just one concern. if python does not support model only generation, shall we report error here?

**Review by tadelesh — APPROVED** [review]

**Pull request merged**